### PR TITLE
fix: ensure correct icon is reflected for monetization state

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Closes https://github.com/interledger/web-monetization-extension/issues/329.

## Changes proposed in this pull request

- We were not sending `IS_FRAME_MONETIZED` due to a premature optimization (not sending it when no tags), so we weren't updating the state when required.
- Also had some timing issues with not `await`-ing.
- We now use `sender.tab.id` to ensure we set the icon for the right tab, and not override it from a stale message from a different tab (race condition).

As discussed in call today, in future we'll try to make use of a `TabStateService` and use that as source of truth, and reduce reliance on message passing.

### CODE REVIEW

File: src/content/lib/messages.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export const checkWalletAddressUrlCall = async (
 export const startMonetization = async (
   payload: ContentToBackgroundActionPayload[ContentToBackgroundAction.START_MONETIZATION]
 ) => {
+  if (!payload.length) return
```
Comment (sidvishnoi) [inline src/content/lib/messages.ts:24]: This is just me avoiding sending unnecessary message.

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">d6f6f4009aeba44fad5fd4b64e9ca7f0f9fef53f</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/9597880704">Run #9597880704</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/25123330911/artifacts/1620735562">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/25123330911/artifacts/1620735565">Download</a></td></tr>
</tbody>
</table>

File: src/background/services/tabEvents.ts [hunk 2]
```diff
@@ -54,29 +54,24 @@ export class TabEvents {
...
+  ) => {
     const { enabled } = await this.storage.get(['enabled'])
 
-    let iconData = {
-      '34': enabled ? icon34 : iconWarning34,
-      '128': enabled ? icon34 : iconWarning128
-    }
+    let iconData = enabled
+      ? { '34': icon34, '128': icon128 }
+      : { '34': iconWarning34, '128': iconWarning128 }
```
Comment (sidvishnoi) [inline src/background/services/tabEvents.ts:64]: Some unrelated refactors. Sorry, OCD.

File: src/background/services/tabEvents.ts [hunk 3]
```diff
@@ -54,29 +54,24 @@ export class TabEvents {
...
+        ? { '34': iconActive34, '128': iconActive128 }
+        : { '34': iconInactive34, '128': iconInactive128 }
     }
 
-    if (this.browser.action) {
-      await this.browser.action.setIcon({ path: iconData })
-    } else if (chrome.browserAction) {
-      chrome.browserAction.setIcon({ path: iconData })
-    }
+    await this.browser.action.setIcon({ path: iconData, tabId })
```
Comment (sidvishnoi) [inline src/background/services/tabEvents.ts:75]: MV3 only now.

Comment (sidvishnoi) [inline src/background/services/tabEvents.ts:75]: MV3 only now.

File: src/background/services/background.ts [hunk 4]
```diff
@@ -131,7 +130,10 @@ export class Background {
               )
 
             case ContentToBackgroundAction.IS_TAB_MONETIZED:
-              this.tabEvents.onUpdatedTab(message.payload)
+              await this.tabEvents.onUpdatedTab({
+                ...message.payload,
+                tabId: sender.tab!.id!
+              })
```
Comment (raducristianpopa) [inline src/background/services/background.ts:136]: Instead of passing the `tabId` directly, let's just pass in the sender and use the `getTabId` utility, just like we do in the monetization service:

https://github.com/interledger/web-monetization-extension/blob/074112917a157fa956796bfbfc2c7b59909ae621/src/background/services/monetization.ts#L114


```suggestion
              await this.tabEvents.onUpdatedTab(message.payload, sender)
```

**Review by raducristianpopa — APPROVED** [review]
